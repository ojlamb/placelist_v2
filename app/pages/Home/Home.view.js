import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';

export const styles = {
  background: {
    backgroundImage: `url(${require('./static/5.jpg')})`,
    position: 'fixed',
    width: '100%',
    height: '100%',
    backgroundSize: 'cover',
    marginTop: 5,

  },
  hero: {
    textAlign: 'center',
    padding: 20,
    backgroundColor: 'rga(250, 250, 250, 0.5)',
  },
  link: {
    textDecoration: 'none',
    color: 'white'
  },
  button: {
    margin: 10
  }
};

export class Home extends React.Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  render() {
    const { classes, session } = this.props;
    return (
      <div className={classes.background}>
        <div className={classes.hero}>
          <h1>Welcome to PlaceList</h1>
          <p>List, map, share and rank your spots</p>
          {session.active ? (
            <Link style={{ textDecoration: 'none' }} to="/places">
              <Button variant="contained" color="primary">
              Explore the List
              </Button>
            </Link>
          ) :
            (
              <div>
                <div>
                  <Link className={classes.link} to="/login">
                    <Button className={classes.button} variant="contained" color="primary">
                    Login
                    </Button>
                  </Link>
                  <Link className={classes.link} to="/signup">
                    <Button className={classes.button} variant="contained" color="primary">Sign Up</Button>
                  </Link>
                </div>
              </div>
            )
        }
        </div>
      </div>
    );
  }
}

Home.propTypes = {
  classes: PropTypes.object.isRequired,
  session: PropTypes.object.isRequired
};
export default withStyles(styles)(Home);
