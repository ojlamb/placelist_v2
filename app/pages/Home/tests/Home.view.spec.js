import React from 'react';
import { Home } from '../Home.view';

const getProps = ({
  session
} = {}) => ({
  classes: {},
  session
});

const getJSX = props => (<Home {...getProps(props)} />);

describe('HeaderView', () => {
  describe('snapshots', () => {
    const scenarios = [
      { session: { active: false, user: {} } },
      { session: { active: true, user: { jwt: 'FOO' } } }
    ];

    scenarios.forEach((props, i) => {
      describe(`Scenario ${i + 1}`, () => {
        it('matches snapshot', () => {
          const wrapper = shallow(getJSX(props));
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
});
