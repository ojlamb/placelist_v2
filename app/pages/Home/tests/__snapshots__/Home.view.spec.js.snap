// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderView snapshots Scenario 1 matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    classes={Object {}}
    session={
      Object {
        "active": false,
        "user": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1>
          Welcome to PlaceList
        </h1>
        <p>
          List, map, share and rank your spots
        </p>
        <div>
          <div>
            <Link
              replace={false}
              to="/login"
            >
              <WithStyles(Button)
                color="primary"
                variant="contained"
              >
                Login
              </WithStyles(Button)>
            </Link>
            <Link
              replace={false}
              to="/signup"
            >
              <WithStyles(Button)
                color="primary"
                variant="contained"
              >
                Sign Up
              </WithStyles(Button)>
            </Link>
          </div>
        </div>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to PlaceList
          </h1>,
          <p>
            List, map, share and rank your spots
          </p>,
          <div>
            <div>
              <Link
                replace={false}
                to="/login"
              >
                <WithStyles(Button)
                  color="primary"
                  variant="contained"
                >
                  Login
                </WithStyles(Button)>
              </Link>
              <Link
                replace={false}
                to="/signup"
              >
                <WithStyles(Button)
                  color="primary"
                  variant="contained"
                >
                  Sign Up
                </WithStyles(Button)>
              </Link>
            </div>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to PlaceList",
          },
          "ref": null,
          "rendered": "Welcome to PlaceList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "List, map, share and rank your spots",
          },
          "ref": null,
          "rendered": "List, map, share and rank your spots",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <Link
                replace={false}
                to="/login"
              >
                <WithStyles(Button)
                  color="primary"
                  variant="contained"
                >
                  Login
                </WithStyles(Button)>
              </Link>
              <Link
                replace={false}
                to="/signup"
              >
                <WithStyles(Button)
                  color="primary"
                  variant="contained"
                >
                  Sign Up
                </WithStyles(Button)>
              </Link>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/login"
                >
                  <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Login
                  </WithStyles(Button)>
                </Link>,
                <Link
                  replace={false}
                  to="/signup"
                >
                  <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Sign Up
                  </WithStyles(Button)>
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Login
                  </WithStyles(Button)>,
                  "className": undefined,
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Login",
                    "className": undefined,
                    "color": "primary",
                    "variant": "contained",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Sign Up
                  </WithStyles(Button)>,
                  "className": undefined,
                  "replace": false,
                  "to": "/signup",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Up",
                    "className": undefined,
                    "color": "primary",
                    "variant": "contained",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
            Welcome to PlaceList
          </h1>
          <p>
            List, map, share and rank your spots
          </p>
          <div>
            <div>
              <Link
                replace={false}
                to="/login"
              >
                <WithStyles(Button)
                  color="primary"
                  variant="contained"
                >
                  Login
                </WithStyles(Button)>
              </Link>
              <Link
                replace={false}
                to="/signup"
              >
                <WithStyles(Button)
                  color="primary"
                  variant="contained"
                >
                  Sign Up
                </WithStyles(Button)>
              </Link>
            </div>
          </div>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Welcome to PlaceList
            </h1>,
            <p>
              List, map, share and rank your spots
            </p>,
            <div>
              <div>
                <Link
                  replace={false}
                  to="/login"
                >
                  <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Login
                  </WithStyles(Button)>
                </Link>
                <Link
                  replace={false}
                  to="/signup"
                >
                  <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Sign Up
                  </WithStyles(Button)>
                </Link>
              </div>
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to PlaceList",
            },
            "ref": null,
            "rendered": "Welcome to PlaceList",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "List, map, share and rank your spots",
            },
            "ref": null,
            "rendered": "List, map, share and rank your spots",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                <Link
                  replace={false}
                  to="/login"
                >
                  <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Login
                  </WithStyles(Button)>
                </Link>
                <Link
                  replace={false}
                  to="/signup"
                >
                  <WithStyles(Button)
                    color="primary"
                    variant="contained"
                  >
                    Sign Up
                  </WithStyles(Button)>
                </Link>
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <WithStyles(Button)
                      color="primary"
                      variant="contained"
                    >
                      Login
                    </WithStyles(Button)>
                  </Link>,
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    <WithStyles(Button)
                      color="primary"
                      variant="contained"
                    >
                      Sign Up
                    </WithStyles(Button)>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(Button)
                      color="primary"
                      variant="contained"
                    >
                      Login
                    </WithStyles(Button)>,
                    "className": undefined,
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Login",
                      "className": undefined,
                      "color": "primary",
                      "variant": "contained",
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(Button)
                      color="primary"
                      variant="contained"
                    >
                      Sign Up
                    </WithStyles(Button)>,
                    "className": undefined,
                    "replace": false,
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign Up",
                      "className": undefined,
                      "color": "primary",
                      "variant": "contained",
                    },
                    "ref": null,
                    "rendered": "Sign Up",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`HeaderView snapshots Scenario 2 matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    classes={Object {}}
    session={
      Object {
        "active": true,
        "user": Object {
          "jwt": "FOO",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1>
          Welcome to PlaceList
        </h1>
        <p>
          List, map, share and rank your spots
        </p>
        <Link
          replace={false}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          to="/places"
        >
          <WithStyles(Button)
            color="primary"
            variant="contained"
          >
            Explore the List
          </WithStyles(Button)>
        </Link>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to PlaceList
          </h1>,
          <p>
            List, map, share and rank your spots
          </p>,
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/places"
          >
            <WithStyles(Button)
              color="primary"
              variant="contained"
            >
              Explore the List
            </WithStyles(Button)>
          </Link>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to PlaceList",
          },
          "ref": null,
          "rendered": "Welcome to PlaceList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "List, map, share and rank your spots",
          },
          "ref": null,
          "rendered": "List, map, share and rank your spots",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
              variant="contained"
            >
              Explore the List
            </WithStyles(Button)>,
            "replace": false,
            "style": Object {
              "textDecoration": "none",
            },
            "to": "/places",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Explore the List",
              "color": "primary",
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Explore the List",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
            Welcome to PlaceList
          </h1>
          <p>
            List, map, share and rank your spots
          </p>
          <Link
            replace={false}
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/places"
          >
            <WithStyles(Button)
              color="primary"
              variant="contained"
            >
              Explore the List
            </WithStyles(Button)>
          </Link>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Welcome to PlaceList
            </h1>,
            <p>
              List, map, share and rank your spots
            </p>,
            <Link
              replace={false}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/places"
            >
              <WithStyles(Button)
                color="primary"
                variant="contained"
              >
                Explore the List
              </WithStyles(Button)>
            </Link>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to PlaceList",
            },
            "ref": null,
            "rendered": "Welcome to PlaceList",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "List, map, share and rank your spots",
            },
            "ref": null,
            "rendered": "List, map, share and rank your spots",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                color="primary"
                variant="contained"
              >
                Explore the List
              </WithStyles(Button)>,
              "replace": false,
              "style": Object {
                "textDecoration": "none",
              },
              "to": "/places",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Explore the List",
                "color": "primary",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Explore the List",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
